homework/publications.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABC, abstractmethod
+from models import BookModel, MagazineModel
+
+
+class AbstractPublication(ABC):
+    @abstractmethod
+    def display_info(self):
+        raise NotImplementedError
+
+
+class Book(AbstractPublication):
+    def __init__(self, model: BookModel):
+        self._model = model
+
+    def __str__(self):
+        return f'Publication: {self._model.name} - {self._model.author}, year - {self._model.year_of_publish}'
+
+    def display_info(self):
+        return self.__str__()
+
+    def to_dict(self):
+        return self._model.dict()
```

Comment:
I think pydantic v2 has method model_dump() for dict representation

Reply:
@marcustas Thanks! Fixed - changed self._model.dict() to self._model.model_dump().